frontend/NotificationViewer.vue

```diff
@@ -0,0 +1,294 @@
+        role="alert"
+    >
+      {{ error }}
+    </div>
+  </li>
+</template>
+
+<script>
+
+export default {
+  name: 'NotificationViewer',
+  props: {
+    url: {
+      type: String,
+      default: '',
+    },
+    interval: {
+      type: Number,
+      default: 300,
+    },
+  },
+  data() {
+    return {
+      notifications: [],
+      error: '',
+    };
+  },
+  computed: {
+    notReadNotificationsCount: function () {
+      return this.notifications.filter(notification => notification.state === 'SENT_STATE').length;
+    },
+  },
+  async created() {
+    await this.fetchNotifications();
+    this.timer = setInterval(this.fetchNotifications, this.interval * 1000);
+  },
+  mounted() {
+    // This next line let the dropdown menu open after clicking inside it. See the bootstrap source code here:
+    // https://github.com/twbs/bootstrap/blob/0b9c4a4007c44201dce9a6cc1a38407005c26c86/js/dropdown.js#L160
+    jQuery(document).on('click.bs.dropdown.data-api', '.notification-dropdown', e => e.stopPropagation());
+    // activate the tooltips
+    jQuery(() => jQuery('[data-toggle="tooltip"]').tooltip());
+  },
+  methods: {
+    fetchNotifications: async function () {
+      try {
+        if (this.url) {
+          const response = await fetch(this.url);
+          let newNotifications = await response.json();
+          this.notifications = newNotifications.results;
+        }
+      } catch (error) {
+        this.error = this.$t('error_fetch_notifications').concat(' (', error.statusText, ')');
+      }
+    },
+    markAsRead: async function (notification) {
+      try {
+        const response = await fetch(this.url.concat(notification.uuid), {
+          method: 'PATCH',
+          headers: {'X-CSRFToken': this.getCookie('csrftoken')},
+        });
+        if (response.status === 200) {
+          const notificationIndex = this.notifications.indexOf(notification);
+          let newNotification = await response.json();
+          this.$set(this.notifications, notificationIndex, newNotification);
+          // Change the bootstrap input radio tooltip
+          jQuery('#'.concat(notification.uuid))
+              .attr('data-original-title', newNotification.state === 'SENT_STATE' ? this.$t('mark_as_read') : this.$t('mark_as_unread'));
+        }
+      } catch (error) {
+        this.error = this.$t('error_mark_as_read').concat(' (', error.statusText, ')');
+      }
+    },
+    markAllAsRead: async function () {
+      try {
+        const response = await fetch(this.url.concat('mark_all_as_read'), {
+          method: 'PUT',
+          headers: {'X-CSRFToken': this.getCookie('csrftoken')},
+        });
+        let notifications = await response.json();
+        if (response.status === 200 && notifications.length > 0) {
+          this.notifications = notifications;
+          // Change ALL the bootstrap input radio tooltips
+          jQuery('[data-toggle="tooltip"]').attr('data-original-title', this.$t('mark_as_unread'));
```

Comment:
Alors tu utilise un moyen détourné de mettre à jour, normalement c'est le title qui fait foi, donc c'est pour ça que tu as un diff entre `title` (le tooltip system) et le tooltip bootstrap. Il vaut mieux mettre à jour les titles des notifications

Reply:
Alors avec `title` ça ne fonctionne juste pas du tout :/ J'ai remplacé ça dans le template pour utiliser `data-original-title` à la place, partout.

Reply:
Alors en effet `data-original-title` semble être la source du tooltip, mais bootstrap set le `title` à rien une fois qu'il l'a récupéré (et "déplacé" dans `data-original-title`) cf https://github.com/twbs/bootstrap/blob/0b9c4a4007c44201dce9a6cc1a38407005c26c86/js/tooltip.js#L348

Le problème, c'est que ton code VueJS rend dynamique le `title`, et celui-ci est donc "reset" à chaque render (engendrant la fameuse diff system<->tooltip), donc je propose que tu contrôle directement dans ton template `data-original-title`, et tu n'as plus besoin des lignes suivantes :
- `:title="notification.state === 'SENT_STATE' ? $t('mark_as_read') : $t('mark_as_unread')"`
- `jQuery('[data-toggle="tooltip"]').attr('data-original-title', this.$t('mark_as_unread'));`